### DISCUSSION TITLE
ASP.NET Core updates in .NET 9 Preview 4

### DISCUSSION THREAD

**Discussion by danroth27** [dotnet/aspnetcore#55820] [Announcements]
Here's a summary of what's new in ASP.NET Core in this preview release:

- [Built-in support for OpenAPI document generation](https://github.com/dotnet/core/blob/main/release-notes/9.0/preview/preview4/aspnetcore.md#built-in-support-for-openapi-document-generation)
- [Introducing `HybridCache`](https://github.com/dotnet/core/blob/main/release-notes/9.0/preview/preview4/aspnetcore.md#introducing-hybridcache)
- [Developer exception page improvements](https://github.com/dotnet/core/blob/main/release-notes/9.0/preview/preview4/aspnetcore.md#developer-exception-page-improvements)
- [Added `OverscanCount` parameter to `QuickGrid`](https://github.com/dotnet/core/blob/main/release-notes/9.0/preview/preview4/aspnetcore.md#added-overscancount-parameter-to-quickgrid)
- [Add static SSR pages to a globally-interactive Blazor Web app](https://github.com/dotnet/core/blob/main/release-notes/9.0/preview/preview4/aspnetcore.md#add-static-ssr-pages-to-a-globally-interactive-blazor-web-app)
- [Avoid 503's during app recycle in IIS](https://github.com/dotnet/core/blob/main/release-notes/9.0/preview/preview4/aspnetcore.md#avoid-503s-during-app-recycle-in-iis)

ASP.NET Core updates in .NET 9 Preview 4:

- [What's new in ASP.NET Core in .NET 9](https://learn.microsoft.com/aspnet/core/release-notes/aspnetcore-9.0) documentation.
- [Breaking changes](https://docs.microsoft.com/dotnet/core/compatibility/9.0#aspnet-core)
- [Roadmap](https://aka.ms/aspnet/roadmap)

.NET 9 Preview 4:

- [Discussion](https://aka.ms/dotnet/9/preview4)
- [Release notes](https://github.com/dotnet/core/blob/main/release-notes/9.0/preview/preview4/README.md)

**Comment by MarvinKlein1508** [comment]
opt-out of Interactive is rendering for certain pages is a huge step! Was waiting for this since .NET 3.1. Much love to the folks at the Blazor team!
